File: app/src/main/java/com/project/scratchstudio/kith_andoid/Activities/AgreementActivity.java
package com.project.scratchstudio.kith_andoid.Activities;

import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.os.SystemClock;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.KeyEvent;
import android.view.View;
import android.webkit.WebView;
import android.widget.Toast;

import com.project.scratchstudio.kith_andoid.R;

public class AgreementActivity extends AppCompatActivity {

    private static long buttonCount = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_agreement);
        WebView webView;
        webView = findViewById(R.id.help_webview);
        webView.getSettings().setJavaScriptEnabled(true);
        if(isNetworkConnected())
            webView.loadUrl("http://kith.ga/terms");
        else Toast.makeText(this, "Нет подключения к интернету", Toast.LENGTH_SHORT).show();
    }

    public void onClickBackButton(View view) {
        view.setEnabled(false);
        finish();
    }

    private boolean isNetworkConnected() {
        ConnectivityManager cm = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
        return cm.getActiveNetworkInfo() != null;
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event)  {
        if (keyCode == KeyEvent.KEYCODE_BACK ) {
            finish();
            return true;
        }
        return super.onKeyDown(keyCode, event);
    }
}

File: app/src/main/java/com/project/scratchstudio/kith_andoid/activities/AgreementActivity.kt
package com.project.scratchstudio.kith_andoid.activities

import android.content.Context
import android.net.ConnectivityManager
import android.os.Bundle
import android.view.KeyEvent
import android.view.View
import android.webkit.WebView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.project.scratchstudio.kith_andoid.R

class AgreementActivity : AppCompatActivity() {

    private val isNetworkConnected: Boolean
        get() {
            val cm = getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            return cm.activeNetworkInfo != null
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_agreement)
        val webView: WebView = findViewById(R.id.help_webview)
        webView.settings.javaScriptEnabled = true
        if (isNetworkConnected)
            webView.loadUrl("http://kith.ga/terms")
        else
            Toast.makeText(this, "Нет подключения к интернету", Toast.LENGTH_SHORT).show()
    }

    fun onClickBackButton(view: View) {
        view.isEnabled = false
        finish()
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        if (keyCode == KeyEvent.KEYCODE_BACK) {
            finish()
            return true
        }
        return super.onKeyDown(keyCode, event)
    }

    companion object {

        private val buttonCount: Long = 0
    }
}
